pkg/artifactory/resource/repository/remote/remote.go

```diff
@@ -618,3 +619,19 @@ func ResourceStateUpgradeV1(_ context.Context, rawState map[string]interface{},
 
 	return rawState, nil
 }
+
+func verifyRemoteRepoLayoutRef(_ context.Context, diff *schema.ResourceDiff, _ interface{}) error {
+	ref := diff.Get("remote_repo_layout_ref").(string)
+	isChanged := diff.HasChange("remote_repo_layout_ref")
+
+	if isChanged && len(ref) == 0 {
+		return fmt.Errorf("empty remote_repo_layout_ref will not remove the actual attribute value and will be ignored by the API, " +
+			"thus will create a state drift on the next plan. Please add the attribute, according to the repository type")
+	}
+
+	return nil
+}
+
+func (r RepositoryRemoteBaseParams) GetRclass() string {
```

Comment:
Not a blocker. Let's move this to right under the `RepositoryRemoteBaseParams` struct definition